crates/core/src/factories/core_fetch.rs

```diff
@@ -0,0 +1,355 @@
+            if !is_agent_cooling_down {
+                // Send fetch request to agent.
+                let peer = match peer_store.get(agent_id.clone()).await {
+                    Ok(Some(peer)) => match &peer.url {
+                        Some(url) => url.clone(),
+                        None => {
+                            tracing::warn!(
+                                "agent {agent_id} no longer in peer store"
+                            );
+                            continue;
+                        }
+                    },
+                    Ok(None) => {
+                        tracing::warn!(
+                            "could not find agent id {agent_id} in peer store"
+                        );
+                        continue;
+                    }
+                    Err(err) => {
+                        tracing::warn!("could not get agent id {agent_id} from peer store: {err}");
+                        continue;
+                    }
+                };
+
+                let mut data = bytes::BytesMut::new();
+                data.put(op_id.clone().0 .0);
+                data.put(agent_id.clone().0 .0);
+                let data = data.freeze();
+
+                if let Err(err) = transport
+                    .send_module(
+                        peer,
+                        space_id.clone(),
+                        MOD_NAME.to_string(),
+                        data,
+                    )
+                    .await
+                {
+                    tracing::warn!("could not send fetch request for op {op_id} to agent {agent_id}: {err}");
+                    state
+                        .lock()
+                        .unwrap()
+                        .cool_down_list
+                        .add_agent(agent_id.clone());
+                }
+            }
+
+            // Re-insert the fetch request into the queue.
+            if let Err(err) =
+                fetch_request_tx.try_send((op_id.clone(), agent_id.clone()))
+            {
+                tracing::warn!("could not re-insert fetch request for op {op_id} to agent {agent_id} in queue: {err}");
+            }
+        }
+    }
+}
+
+impl Drop for Inner {
+    fn drop(&mut self) {
+        for t in self.fetch_tasks.iter() {
+            t.abort();
+        }
+    }
+}
+
+#[derive(Debug)]
+struct CoolDownList {
+    state: HashMap<AgentId, Instant>,
+    cool_down_interval: u64,
+}
+
+impl CoolDownList {
+    pub fn new(cool_down_interval: u64) -> Self {
+        Self {
+            state: HashMap::new(),
+            cool_down_interval,
+        }
+    }
+
+    pub fn add_agent(&mut self, agent_id: AgentId) {
+        self.state.insert(agent_id, Instant::now());
+    }
+
+    pub fn is_agent_cooling_down(&mut self, agent_id: &AgentId) -> bool {
+        match self.state.get(agent_id) {
+            Some(instant) => {
+                let now = Instant::now();
+                if (now - *instant).as_millis()
```

Comment:
This can just use https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed

Reply:
done https://github.com/holochain/kitsune2/pull/35/commits/9ac37438280fe7e14884b933d4d38a901aca2ab8